software/base/src/main/java/org/apache/brooklyn/entity/software/base/lifecycle/MachineLifecycleEffectorTasks.java

```diff
@@ -618,7 +620,16 @@ public void restart(ConfigBag parameters) {
         if (isRestartMachine==RestartMachineMode.FALSE) {
             DynamicTasks.queue("stopping (process)", new StopProcessesAtMachineTask());
         } else {
-            DynamicTasks.queue("stopping (machine)", new StopMachineTask());
+            Map<String, Object> stopMachineFlags =  MutableMap.of();
+            if (Entitlements.getEntitlementContext() != null) {
+                stopMachineFlags.put("tags", MutableSet.of(BrooklynTaskTags.tagForEntitlement(Entitlements.getEntitlementContext())));
```

Comment:
Again this feels like a fix for just one situation. Presumably there are lots of places that have an effector/task that queues sub-tasks. Those other places may not inherit the EntitlementContext, if we only fix it here.

But saying that, if the DynamicTasks.queue ends up submitting it via `BasicExecutionContext.submitInternal` then it will happen automatically (I think).